hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -155,8 +163,9 @@ type HostedClusterReconciler struct {
 
 	MetricsSet metrics.MetricsSet
 
-	overwriteReconcile func(ctx context.Context, req ctrl.Request, log logr.Logger, hcluster *hyperv1.HostedCluster) (ctrl.Result, error)
-	now                func() metav1.Time
+	overwriteReconcile    func(ctx context.Context, req ctrl.Request, log logr.Logger, hcluster *hyperv1.HostedCluster) (ctrl.Result, error)
+	now                   func() metav1.Time
+	kubevirtInfraClusters sync.Map
```

Comment:
maybe this should be called `kubevirtInfraClients` instead?  the issue with kubevirtInfraClusters is might be confused with kubevirtCLuster objects